src/optimizer/optimizer.cpp

```diff
@@ -98,19 +101,18 @@ std::unique_ptr<planner::AbstractPlanNode> Optimizer::ChooseBestPlan(
       child_expr_map[input_cols[i][offset]] = offset;
     }
 
-    auto child_plan = ChooseBestPlan(txn, accessor, child_groups[i], required_input_props[i], input_cols[i]);
+    auto child_plan = ChooseBestPlan(txn, accessor, child_groups[i], required_input_props[i], input_cols[i], generator);
     NOISEPAGE_ASSERT(child_plan != nullptr, "child should have derived a non-null plan...");
-
     children_plans.emplace_back(std::move(child_plan));
     children_expr_map.push_back(child_expr_map);
   }
 
   // Derive root plan
   auto *op = new OperatorNode(gexpr->Contents(), {}, txn);
 
-  PlanGenerator generator;
-  auto plan = generator.ConvertOpNode(txn, accessor, op, required_props, required_cols, output_cols,
-                                      std::move(children_plans), std::move(children_expr_map));
+  planner::PlanMetaData::PlanNodeMetaData plan_node_meta_data(context_->GetMemo().GetGroupByID(id)->GetNumRows());
+  auto plan = generator->ConvertOpNode(txn, accessor, op, required_props, required_cols, output_cols,
+                                       std::move(children_plans), std::move(children_expr_map), plan_node_meta_data);
```

Comment:
You can move the `plan_node_meta_data` into the function, similar to other ephemeral objects.